src/atoms/TextInput/TextInput.web.js

```diff
@@ -1,3 +1,486 @@
+    }, 90);
+  }, []);
+
+  const onBlurText = useCallback(
+    (event) => {
+      setIsFocused(false);
+      setIsPlaceholderVisible(false);
+      const text = event.target.value;
+      setInput(text);
+      if (onBlur) {
+        onBlur(text);
+      }
+    },
+    [setIsFocused, setIsPlaceholderVisible, setInput, onBlur],
+  );
+
+  const onChangeText = useCallback(
+    (event) => {
+      const text = event.target.value;
+      if (text.length > maxLength) {
+        return;
+      }
+      setInput(text);
+      if (onChange) {
+        onChange(text);
+      }
+    },
+    [maxLength, onChange, setInput],
+  );
+
+  const onSelectText = () => {
+    return theme.colors.primary[980];
+  };
+
+  const onKeyPress = (e) => {
+    if (type === 'number') {
+      const charCode = typeof e.which === 'number' ? e.which : e.keyCode;
+      const char = String.fromCharCode(charCode);
+      const isAllowed = /[0-9]/g.test(char) || char === '-' || char === '.';
+
+      if (!isAllowed) {
+        e.preventDefault();
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (isDefined(value)) {
+      setInput(value);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    if (inputRef.current && containerRef.current) {
+      const containerRect = containerRef.current.getBoundingClientRect();
+      const inputRect = inputRef.current.getBoundingClientRect();
+      const initialLeftPosition = inputRect.x - containerRect.x;
+      const finalTopPosition = inputRef.current.offsetHeight;
+
+      setLayoutDimensions({
+        initialLeftPosition,
+        finalTopPosition,
+      });
+    }
+  }, [inputRef, containerRef]);
+
+  return (
+    <Flex justifyContent="flex-end" flexDirection="column">
+      <View ref={containerRef}>
+        {!hasAnimatedLabel && labelPosition === 'top' ? (
+          <Label.Regular
+            position={labelPosition}
+            disabled={disabled}
+            isFocused={isFocused}
+            variant={variant}
+            hasError={hasError}
+            hasText={hasText}
+            value={input}
+            width={width}
+            id={id}
+          >
+            {label}
+          </Label.Regular>
+        ) : null}
+
+        {/* Text Input Container */}
+        <Flex flexDirection="row" alignItems="flex-start">
+          <View>
+            {!hasAnimatedLabel && labelPosition === 'left' ? (
+              <Label.Regular
+                position={labelPosition}
+                disabled={disabled}
+                isFocused={isFocused}
+                variant={variant}
+                hasError={hasError}
+                hasText={hasText}
+                value={input}
+                width={width}
+                id={id}
+              >
+                {label}
+              </Label.Regular>
+            ) : null}
+            {/* Text Input */}
+            <Flex flexDirection="column" flex={width === 'auto' ? 1 : 0}>
+              <View>
+                <Size height="36px" minHeight="auto">
```

Comment:
```suggestion
                <Size height={4.5} minHeight="auto">
```

Reply:
changed in https://github.com/razorpay/blade/pull/160/files#diff-e894a2dc2387b5d3ed36a8c505fba9deR54